SYCL/ESIMD/api/functional/ctors/ctor_default.cpp

```diff
@@ -78,57 +74,56 @@ struct const_ref {
   }
 
   template <typename DataT, int NumElems>
-  static void call_simd_ctor(DataT *output_data) {
+  static void call_simd_ctor(DataT *const output_data) {
     call_simd_by_const_ref<DataT, NumElems>(simd<DataT, NumElems>(),
                                             output_data);
   }
 };
 
-// Functor with the main test routine to iterate over the pre-determined
-// datatypes
-template <typename DataT, int NumElems, typename TestCaseT> struct test {
+// Struct that calls simd in provided context and then verifies obtained result.
+template <typename DataT, int NumElems, typename TestCaseT> struct run_test {
   bool operator()(sycl::queue &queue, const std::string &data_type) {
-    bool passed{true};
-
+    bool passed = true;
     DataT default_val{};
-    using AllocatorT = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
 
-    std::vector<DataT, AllocatorT> result{NumElems, AllocatorT{queue}};
+    shared_vector<DataT> result(NumElems, shared_allocator<DataT>(queue));
+
     queue.submit([&](sycl::handler &cgh) {
-      auto out = result.data();
-      cgh.single_task<Kernel<DataT, NumElems, TestCaseT>>(
+      DataT *const out = result.data();
+      cgh.single_task<ctors::Kernel<DataT, NumElems, TestCaseT>>(
           [=]() SYCL_ESIMD_KERNEL {
             TestCaseT::template call_simd_ctor<DataT, NumElems>(out);
           });
     });
-    for (const auto &it : result) {
-      if (it != default_val) {
+
+    for (size_t i = 0; i < result.size(); ++i) {
+      if (result[i] != default_val) {
         passed = false;
-        log::fail<NumElems>(
-            "In simd by " + TestCaseT::get_description() +
-                " elem value is not equal to default value, retrieved: " +
-                std::to_string(it) +
-                ", expected: " + std::to_string(default_val),
-            data_type);
+
+        const auto description =
+            ctors::TestDescription<DataT, NumElems, TestCaseT>(
+                i, result[i], default_val, data_type);
+        log::fail(description);
       }
     }
+
     return passed;
   }
 };
 
-int main(int argc, char **argv) {
-  sycl::queue queue{esimd_test::ESIMDSelector{},
-                    esimd_test::createExceptionHandler()};
+int main(int, char **) {
+  sycl::queue queue(esimd_test::ESIMDSelector(),
```

Comment:
```suggestion
  sycl::queue queue(esimd_test::ESIMDSelector{},
```
Minor, actually. Looks like all other tests use such way to initialize the ESIMDSelector instance, probably to visually differentiate the instantiation from the function call. What do you think? Here and in ctor_copy.

Reply:
I've done it in this commit: 06d104d6498d91c00d7acd83e697c6c9fd5c0856